warehouse/slave_worker.go

```diff
@@ -0,0 +1,504 @@
+
+		jr.cleanup()
+	}()
+
+	var (
+		err                  error
+		lineBytesCounter     int
+		interfaceSliceSample []interface{}
+	)
+
+	if jr.stagingFilePath, err = jr.getStagingFilePath(sw.workerIdx); err != nil {
+		return nil, err
+	}
+	if err = jr.downloadStagingFile(ctx); err != nil {
+		return nil, err
+	}
+	if jr.stagingFileReader, err = jr.reader(); errors.Is(err, io.EOF) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	jr.uuidTS = jr.now()
+
+	// Initialize Discards Table
+	discardsTable := job.discardsTable()
+	jr.tableEventCountMap[discardsTable] = 0
+
+	processingStart := jr.now()
+	sortedTableColumnMap := job.sortedColumnMapForAllTables()
+
+	// default scanner buffer maxCapacity is 64K
+	// set it to higher value to avoid read stop on read size error
+	maxCapacity := sw.config.maxStagingFileReadBufferCapacityInK * 1024
+
+	bufScanner := bufio.NewScanner(jr.stagingFileReader)
+	bufScanner.Buffer(make([]byte, maxCapacity), maxCapacity)
+
+	for {
+		ok := bufScanner.Scan()
+		if !ok {
+			if scanErr := bufScanner.Err(); scanErr != nil {
+				jr.logger.Errorf("WH: Error in scanner reading line from staging file: %v", scanErr)
+			}
+			break
+		}
+
+		lineBytes := bufScanner.Bytes()
+		lineBytesCounter += len(lineBytes)
+
+		var (
+			batchRouterEvent BatchRouterEvent
+			writer           encoding.LoadFileWriter
+		)
+
+		if err := json.Unmarshal(lineBytes, &batchRouterEvent); err != nil {
+			jr.logger.Errorf("[WH]: Failed to unmarshal JSON line to batchrouter event: %+v", batchRouterEvent)
+			continue
+		}
+
+		tableName := batchRouterEvent.Metadata.Table
+		columnData := batchRouterEvent.Data
+
+		if job.DestinationType == warehouseutils.S3Datalake && len(sortedTableColumnMap[tableName]) > columnCountLimitMap[warehouseutils.S3Datalake] {
+			return nil, fmt.Errorf("staging file schema limit exceeded for stagingFileID: %d, actualCount: %d",
+				job.StagingFileID,
+				len(sortedTableColumnMap[tableName]),
+			)
+		}
+
+		// Create separate load file for each table
+		if writer, err = jr.writer(tableName); err != nil {
+			return nil, err
+		}
+
+		eventLoader := encoding.GetNewEventLoader(job.DestinationType, job.LoadFileType, writer)
+
+		for _, columnName := range sortedTableColumnMap[tableName] {
+			if eventLoader.IsLoadTimeColumn(columnName) {
+				timestampFormat := eventLoader.GetLoadTimeFormat(columnName)
+				eventLoader.AddColumn(job.columnName(columnName), job.UploadSchema[tableName][columnName], jr.uuidTS.Format(timestampFormat))
+				continue
+			}
+
+			columnInfo, ok := batchRouterEvent.GetColumnInfo(columnName)
+			if !ok {
+				eventLoader.AddEmptyColumn(columnName)
+				continue
+			}
+
+			columnType := columnInfo.Type
+			columnVal := columnInfo.Value
+
+			if job.DestinationType == warehouseutils.CLICKHOUSE {
+				switch columnType {
+				case "boolean":
```

Comment:
Don't we have constants for these? If not we should create them. A follow up PR is also fine :+1: